src/main/java/de/zalando/zmon/scheduler/ng/cleanup/SingleEntityCleanup.java

```diff
@@ -56,6 +62,9 @@ public void notifyEntityRemove(EntityRepository repo, Entity e) {
         executor.schedule(new EntityCleanupTask(e), 300, TimeUnit.SECONDS);
     }
 
+    @Override
+    public void notifyBatchEntityRemove (EntityRepository repo, Set<String> removedEntities) {}
```

Comment:
I'm not sure, but it sounds semantically incorrect to have batch operation method in a class called `SingleEntityCleanup`. Did you think about introducing new Listener interface? Like `BatchEntityChangeListener`

Reply:
The EntityRepository only allows listeners of type `EntityChangeListener` https://github.com/zalando-zmon/zmon-scheduler/blob/master/src/main/java/de/zalando/zmon/scheduler/ng/entities/EntityRepository.java#L31